packages/ui/ViewManager/ViewManager.js

```diff
@@ -189,33 +189,33 @@ class ViewManager extends React.Component {
 		index: 0
 	}
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		this.previousIndex = this.props.index;
-		this.checkReverse(nextProps);
+	constructor (props) {
+		super(props);
+		this.state = {
+			previousIndex: null,
```

Comment:
minor nit: Let's keep consistency with other changes and name it `prevIndex` to shorten the variable name